classes/services/queryBuilders/PKPUserQueryBuilder.inc.php

```diff
@@ -441,267 +443,227 @@ public function getIds() {
+					switch (\Config::getVar('database', 'driver')) {
+						case 'mysql':
+						case 'mysqli':
+							$dateDiffClause = 'DATEDIFF(ra.date_completed, ra.date_notified)';
+							break;
+						default:
+							$dateDiffClause = "DATE_PART('day', ra.date_completed - ra.date_notified)";
+					}
+					$query->from('review_assignments', 'ra')
+						->groupBy('ra.reviewer_id')
+						->select('ra.reviewer_id')
+						->selectRaw('MAX(ra.date_assigned) AS last_assigned')
+						->selectRaw('COUNT(CASE WHEN ra.date_completed IS NULL AND ra.declined = 0 THEN 1 END) AS incomplete_count')
+						->selectRaw('COUNT(CASE WHEN ra.date_completed IS NOT NULL AND ra.declined = 0 THEN 1 END) AS complete_count')
+						->selectRaw('SUM(ra.declined) AS declined_count')
+						->selectRaw('SUM(ra.cancelled) AS cancelled_count')
+						->selectRaw("AVG($dateDiffClause) AS average_time")
+						->selectRaw('AVG(ra.quality) AS reviewer_rating');
+				}, 'ra_stats', 'u.user_id', '=', 'ra_stats.reviewer_id')
+
+					// Select all statistics columns
+					->addSelect('ra_stats.*')
+
+					// Reviewer rating
+					->when(!empty($this->reviewerRating), function (Builder $query) {
+						$query->where('ra_stats.reviewer_rating', '>=', $this->reviewerRating);
+					})
+
+					// Completed reviews
+					->when(!empty($this->reviewsCompleted), function (Builder $query) {
+						$reviewsCompleted = is_array($this->reviewsCompleted) ? $this->reviewsCompleted : [$this->reviewsCompleted];
+						if (count($reviewsCompleted) > 1) {
+							$query->whereBetween('ra_stats.complete_count', $reviewsCompleted);
+						} else {
+							$query->where('ra_stats.complete_count', '>=', reset($reviewsCompleted));
+						}
+					})
+
+					// Active reviews
+					->when(!empty($this->reviewsActive), function (Builder $query) {
+						$reviewsActive = is_array($this->reviewsActive) ? $this->reviewsActive : [$this->reviewsActive];
+						if (count($reviewsActive) > 1) {
+							$query->whereBetween('ra_stats.incomplete_count', $reviewsActive);
+						} else {
+							$query->where('ra_stats.incomplete_count', '>=', reset($reviewsActive));
+						}
+					})
+
+					// Days since last review assignment
+					->when(!empty($this->daysSinceLastAssignment), function (Builder $query) {
+						$daysSinceLastAssignment = is_array($this->daysSinceLastAssignment) ? $this->daysSinceLastAssignment : [$this->daysSinceLastAssignment];
+						$dbTimeMin = (string) Carbon::now()->subDays((int) reset($daysSinceLastAssignment))->toDateString();
+						$query->where('ra_stats.last_assigned', '<=', $dbTimeMin);
+						if (count($daysSinceLastAssignment) > 1) {
+							// Subtract an extra day so that our outer bound rounds "up". This accounts for the UI rounding "down" in the string "X days ago".
+							$dbTimeMax = (string) Carbon::now()->subDays((int) end($daysSinceLastAssignment) + 1)->toDateString();
+							$query->where('ra_stats.last_assigned', '>=', $dbTimeMax);
+						}
```

Comment:
This whole snippet sounded strange for me when comparing against the `/main` version, so I kept the previous behavior here 🤔

If you don't know which one is right, I'll find where it's used and play with the interface to check whether it's a bug or not.

- Different filter
```php
$query->where('ra_stats.last_assigned', '>', $dbTimeMin); // main
$query->where('ra_stats.last_assigned', '<=', $dbTimeMin);
```

- Main `subs` days, while this branch `adds`
```php
$dbTimeMax = (string) Carbon::now()->subDays((int) end($daysSinceLastAssignment) + 1)->toDateString(); // main
$dbTimeMax = (string) Carbon::now()->addDays((int) end($daysSinceLastAssignment) + 1)->toDateString();
```

Another difference on the filter
```php
$query->where('ra_stats.last_assigned', '<', $dbTimeMax); //main
$query->where('ra_stats.last_assigned', '>=', $dbTimeMax);
```

Reply:
Please run a few tests on it to see if it's behaving right, thanks!

Reply:
Ok, I'll take a look. Just wanted to get a fast answer :)

Reply:
FYI I concluded the `/main` was incorrect.